### PR TITLE
Add basic docs

### PR DESCRIPTION
**Pull request by AyanSinhaMahapatra**
1. Adds Basic Sphinx Documentation files
2. Add RTD configuration and requirements file
3. Test Scripts

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'nexb-skeleton'
+copyright = '2020, nexb Inc.'
```
Comment (majurg) [inline docs/source/conf.py:21]: I think it's best to leave off the Copyright Year in general.

File: docs/source/skeleton/index.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+.. include:: ../../../README.rst
```
Comment (majurg) [inline docs/source/skeleton/index.rst:1]: We should add some comments above this line telling the user to modify this index file content.

File: docs/requirements-doc.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+doc8==0.8.1
+Sphinx==3.3.1
+sphinx-rtd-theme==0.5.0
```
Comment (majurg) [inline docs/requirements-doc.txt:3]: I do not think we need a separate requirements-doc.txt file. We should probably add these here: https://github.com/nexB/skeleton/blob/ef210cd813de2961fe8ca4a5ec7f14532ea6e9f8/pyproject.toml#L2

**Review by majurg — CHANGES_REQUESTED** [review]
@AyanSinhaMahapatra This looks good, just a few minor comments.

File: docs/source/conf.py (hunk 1 above)
Comment (AyanSinhaMahapatra) [reply docs/source/conf.py:21]: Removing yes.

File: docs/requirements-doc.txt (hunk 3 above)
Comment (AyanSinhaMahapatra) [reply docs/requirements-doc.txt:3]: You're right. There seems to be a way to include it [like this](https://docs.readthedocs.io/en/latest/config-file/v1.html#python-extra-requirements). Doing this.

File: docs/source/skeleton/index.rst (hunk 2 above)
Comment (AyanSinhaMahapatra) [reply docs/source/skeleton/index.rst:1]: Sure! Makes more sense.

**Comment by AyanSinhaMahapatra** [comment]
@majurg I'm not sure why the RTD build is failing would you check - https://readthedocs.org/projects/ayan-fork-nexb-skeleton/builds/12558609/. The requirements install fine, local build also passes. Here are the docs I followed - https://docs.readthedocs.io/en/latest/config-file/v2.html#python

**Review by majurg — APPROVED** [review]
@AyanSinhaMahapatra This looks good. 

The error the RTD build returns is a problem. This is because a fresh clone of the `skeleton` repo does not contain a `src/` directory that is usually present in an actual library (like [extractcode](https://github.com/nexB/extractcode)

I think the best approach to solve this issue is to remove this line: https://github.com/nexB/skeleton/blob/ef210cd813de2961fe8ca4a5ec7f14532ea6e9f8/setup.cfg#L23

This will allow the `./configure` script and the RTD CI system to run correctly. Once you remove that line and confirm, I will merge this.

**Comment by AyanSinhaMahapatra** [comment]
Isn't that relevant for the package setup when the src is present? Like users still have to uncomment/use that line when they use the package right?

This is the same error with which the Travis builds for skeleton is failing too. 

Would having a `src` directory here in skeleton, instead, just with a __init__.py or maybe some dummy functions work instead? That would pass both the failing tests and this RTD build maybe?

**Comment by majurg** [comment]
@AyanSinhaMahapatra Yes that is also a possible solution, though I would prefer to just have minimal README file that just says "put your python modules in this directory" instead of a dummy python file.

**Comment by AyanSinhaMahapatra** [comment]
@majurg Yeah, that makes more sense. Do you want me to add a commit doing the same here?

**Comment by AyanSinhaMahapatra** [comment]
[RTD build](https://ayan-fork-nexb-skeleton.readthedocs.io/en/add-basic-docs/) passes, CI tests fail because now even though `src` is there, no tests are collected by `pytest` [exit code 5].

**Comment by majurg** [comment]
@AyanSinhaMahapatra I do not really think it matters that the tests do not pass for this skeleton repo (as it is a template, not actual code).

I have removed auto-builds in travis for now for this repo.

**Pull request merged**